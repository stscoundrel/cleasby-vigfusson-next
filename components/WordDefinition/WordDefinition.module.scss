.section {
  padding: 20px 0;
}

.definitionList {
  margin-bottom: 20px;
}

.itemDescription {
  padding-left: 20px;
}

@media only screen and (min-width: 1200px) {
  .section {
    max-width: 740px;
  }
}