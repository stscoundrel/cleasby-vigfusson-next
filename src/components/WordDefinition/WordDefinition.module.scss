.section {
  padding: 20px 0;
}

.subHeading {
  display: block;
  margin-bottom: 30px;
}

.rune {
  display: block;
  margin-top: 10px;
  font-size: 20px;
}

.definitionList {
  margin-bottom: 20px;
}

.itemDescription {
  padding-left: 20px;
}

@media only screen and (min-width: 768px) {
  .rune {
    display: inline-block;
    font-size: 16px;
    margin-top: 0;
    margin-left: 5px;
  }
}

@media only screen and (min-width: 1200px) {
  .section {
    max-width: 740px;
  }
}