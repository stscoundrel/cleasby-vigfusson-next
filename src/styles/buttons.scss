.button {
  display: block;
  background-color: $color-blue;
  border: 2px solid $color-blue;
  color: white;
  padding: 5px 15px;
  margin: 20px 0;
  min-width: 100px;
  text-align: center;

  &:hover {
    background-color: transparent;
    color: $color-blue;
    text-decoration: none;
  }

  &--secondary {
    background: transparent;
    color: white;
    border: 2px solid white;

    &:hover {
      background-color: white;
      color: $color-blue;
    }
  }

  &--no-margs {
    margin: 0 !important;
  }
}


@media only screen and (min-width: 768px) {
  .button {
    display: inline-block;
    margin: 15px 0;
  }
}