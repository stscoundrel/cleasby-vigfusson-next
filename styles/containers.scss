.container {
  width: 100%;
  padding-right: 16px;
  padding-left: 16px;
  margin-right: auto;
  margin-left: auto;
  max-width: calc(100% - 32px);
}

@media (min-width: 640px) {
  .container {
    max-width: 640px;
  }
}

@media (min-width: 768px) {
  .container {
    padding-right: 35px;
    padding-left: 35px;
  }
}

@media (min-width: 1024px) {
  .container {
    max-width: 800px;
  }
}

@media (min-width: 1280px) {
  .container {
    max-width: 1280px;
  }
}

@media (min-width: 1536px) {
  .container {
    max-width: 1536px;
  }
}