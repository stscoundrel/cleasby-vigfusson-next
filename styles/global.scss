@import '~reset-css';
@import 'variables';
@import 'typography';
@import 'containers';

body,
html {
  max-width: 100%;
  overflow-x: hidden;
  scroll-behavior: smooth;
}

.capitalize {
  text-transform: capitalize;
}