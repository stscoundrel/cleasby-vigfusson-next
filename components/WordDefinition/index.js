import styles from './WordDefinition.module.scss'

export default function WordDefinition({ data }) {
  const { word, definitions } = data

  return (
    <article className={styles.section}>
      <header>
        <h1 className="capitalize" lang="non">{word}</h1>
        <p>Meaning of Old Norse word <em>&quot;{word}&quot;</em></p>
      </header>

      <p><dfn className="capitalize">{word}</dfn> Old Norse word can mean:</p>
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong></dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={ { __html: definition } }
          ></dd>
        </dl>
      ))}
    </article>
  )
}
