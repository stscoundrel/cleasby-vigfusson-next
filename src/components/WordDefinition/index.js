import { capitalize } from 'lib/utils/strings'
import { lettersToRunes } from 'younger-futhark'
import styles from './WordDefinition.module.scss'

export default function WordDefinition({ data }) {
  const { word, definitions } = data

  return (
    <article className={styles.section}>
      <header>
        <h1 className="capitalize" lang="non">{capitalize(word)}</h1>
        <p>Possible runic inscription in <em>Younger Futhark:</em> { lettersToRunes(word) }</p>
        <p>Meaning of Old Norse word <em>&quot;{word}&quot;</em></p>
      </header>

      {definitions.length > 1 && <p><dfn className="capitalize">{word}</dfn> Old Norse word can mean:</p>}
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong></dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={ { __html: definition } }
          ></dd>
        </dl>
      ))}
    </article>
  )
}
