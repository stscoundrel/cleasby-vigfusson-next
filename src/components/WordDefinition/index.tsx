import { youngerFuthark } from 'riimut'
import type { Crosslink } from 'scandinavian-dictionary-crosslinker'
import { capitalize, getOlderSpelling } from 'lib/utils/strings'
import { addAbbreviationsToContent, type CombinedAbbreviations } from 'lib/services/abbreviations'
import Abbreviations from 'components/Abbreviations'
import type { DictionaryEntry } from 'lib/services/dictionary'
import Crosslinks from 'components/Crosslinks'
import SimilarEntries from 'components/SimilarEntries'
import styles from './WordDefinition.module.scss'

interface WordDefinitionProps{
  entry: DictionaryEntry,
  similarEntries: DictionaryEntry[],
  abbreviations: CombinedAbbreviations,
  crosslinks: Crosslink[],
}

export default function WordDefinition({
  entry, similarEntries, abbreviations, crosslinks,
}: WordDefinitionProps) {
  const { word, definitions } = entry
  const olderForm = getOlderSpelling(word)
  const hasOlderForm = word !== olderForm

  return (
    <article className={styles.section}>
      <header>
        <h1 lang="non">{capitalize(word)}</h1>

        <small className={styles.subHeading}>
          Old Norse Dictionary - {word.toLowerCase()}
        </small>
        <p>Meaning of Old Norse word <em>&quot;{word}&quot;</em>
        {hasOlderForm && <> (or <em>{olderForm}</em>)</>} in English.</p>
      </header>

      <p>As defined by the Cleasby & Vigfusson Old Norse to English dictionary:</p>

      {definitions.length > 1 && <p><dfn className="capitalize">{word}</dfn> Old Norse word can mean:</p>}
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong> {hasOlderForm && <>({olderForm})</>}</dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={{
              __html: addAbbreviationsToContent(definition, abbreviations),
            } }
          ></dd>
        </dl>
      ))}

      {hasOlderForm
        && <p>
          <strong>Orthography: </strong>The Cleasby & Vigfusson book used letter <em>ö </em>
          to represent the original Old Norse vowel <em>ǫ</em>. Therefore, <em>{word}</em> may be
          more accurately written as <em>{olderForm}</em>.
        </p>}

      <p>Possible runic inscription in <em>Younger Futhark</em>:
        <span className={styles.rune}>{ youngerFuthark.lettersToRunes(word) }</span><br />
      <small>Younger Futhark runes were used from 8th to 12th centuries
        in Scandinavia and their overseas settlements</small>
      </p>

      <SimilarEntries entries={similarEntries} />
      <Abbreviations abbreviations={abbreviations} />
      <br />
      <Crosslinks crosslinks={crosslinks} />
    </article>
  )
}
