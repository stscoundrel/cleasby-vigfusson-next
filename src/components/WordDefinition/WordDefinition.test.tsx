import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import { DictionarySource } from 'scandinavian-dictionary-crosslinker'
import WordDefinition from './index'
import styles from './WordDefinition.module.scss'

const word = {
  word: 'af-burðr',
  definitions: [
    'm. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Orkn. 464.',
    '<strong>II.</strong> gen. sing. and pl. <strong>afburðar-, a-,</strong> freq. used as a prefix in some COMPDS with the notion of <i>gloriously, with distinction.</i> <strong>afburðar-digr,</strong> adj. <i>very thick</i>, biðr. 24. <strong>afburða-fræknligr,</strong> adj. <i>very gallant,</i> Ísl. ii. 369. <strong>afburðar-járn,</strong> n. <i>excellent iron,</i> Fms. x. 173. <strong>afburðar-maðr,</strong> m. <i>a man of mark,</i> Rb. 316, Orkn. 474, Grett. 133, Finnb. 318. <strong>afburðar-mikill,</strong> adj. <i>conspicuous,</i> Fms. v. 181. <strong>afburðar-skip,</strong> n. <i>a fine ship,</i> Fas. iii. 106. <strong>afburðar-vel,</strong> adv. <i>very well</i>, Hkr. ii. 265, Fms. ix. 515. <strong>afburðar-vænn,</strong> adj. <i>very fine,</i> Fas. i. 182.',
  ],
  slug: 'af-burdr',
}

const wordWithOlderSpellingVariant = {
  word: 'ör-væna',
  definitions: [
    'u, f. anything beyond hope; mér er Sveins á engri stundu örvæna, no hour at which I may not expect Sweyn, Orkn. 412 (örvænt, v. l.); margir menn mæla, at eigi sé örvæna at ek koma þar, that it is not past hope, that …, it may well be that …, Ld. 204.',
  ],
  slug: 'or-vana',
}

const abbreviations = {
  common: [
    {
      abbreviation: 'f.',
      explanation: 'feminine.',
    },
    {
      abbreviation: 'n.',
      explanation: 'neuter.',
    },
  ],
  works: [
    {
      abbreviation: 'Björn',
      explanation: 'Biörn Halldórsson.',
    },
  ],
}

const crosslinks = [
  {
    url: 'https://old-icelandic.vercel.app/word/fadir',
    source: DictionarySource.OldIcelandic,
  },
  {
    url: 'https://old-norwegian-dictionary.vercel.app/word/fadir',
    source: DictionarySource.OldNorwegian,
  },
  {
    url: 'https://old-swedish-dictionary.vercel.app/word/fadhir',
    source: DictionarySource.OldSwedish,
  },
]

describe('WordDefinition component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        similarEntries={[]}
        crosslinks={crosslinks}
      />,
    )
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        similarEntries={[]}
        crosslinks={crosslinks}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Matches snapshot (older spelling variant)', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={wordWithOlderSpellingVariant}
        similarEntries={[]}
        abbreviations={abbreviations}
        crosslinks={crosslinks}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        similarEntries={[]}
        crosslinks={crosslinks}
      />,
    )
    const { root } = tree

    expect(root.findByType('h1').children).toEqual(['Af-burðr'])
  })

  test('Has correct amount of definitions', () => {
    const tree = renderer.create(
      <WordDefinition
        entry={word}
        abbreviations={abbreviations}
        similarEntries={[]}
        crosslinks={crosslinks}
      />,
    )
    const { root } = tree

    expect(root.findAllByProps({ className: styles.definitionList }).length).toEqual(2)
  })
})
