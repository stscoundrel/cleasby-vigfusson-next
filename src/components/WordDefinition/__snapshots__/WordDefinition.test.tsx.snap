// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordDefinition component Matches snapshot (older spelling variant) 1`] = `
<article
  className="section"
>
  <header>
    <h1
      lang="non"
    >
      Ör-væna
    </h1>
    <small
      className="subHeading"
    >
      Old Norse Dictionary - 
      ör-væna
    </small>
    <p>
      Meaning of Old Norse word 
      <em>
        "
        ör-væna
        "
      </em>
       (or 
      <em>
        ǫr-væna
      </em>
      )
       in English.
    </p>
  </header>
  <p>
    As defined by the Cleasby & Vigfusson Old Norse to English dictionary:
  </p>
  <dl
    className="definitionList"
  >
    <dt>
      <strong>
        ör-væna
      </strong>
       
      (
      ǫr-væna
      )
    </dt>
    <dd
      className="itemDescription"
      dangerouslySetInnerHTML={
        {
          "__html": "u, <abbr title="feminine.">f.</abbr> anything beyond hope; mér er Sveins á engri stundu örvæna, no hour at which I may not expect Sweyn, Ork<abbr title="neuter.">n.</abbr> 412 (örvænt, v. l.); margir menn mæla, at eigi sé örvæna at ek koma þar, that it is not past hope, that …, it may well be that …, Ld. 204.",
        }
      }
    />
  </dl>
  <p>
    <strong>
      Orthography: 
    </strong>
    The Cleasby & Vigfusson book used letter 
    <em>
      ö 
    </em>
    to represent the original Old Norse vowel 
    <em>
      ǫ
    </em>
    . Therefore, 
    <em>
      ör-væna
    </em>
     may be more accurately written as 
    <em>
      ǫr-væna
    </em>
    .
  </p>
  <p>
    Possible runic inscription in 
    <em>
      Younger Futhark
    </em>
    :
    <span
      className="rune"
    >
      ᚢᚱ-ᚢᛅᚾᛅ
    </span>
    <br />
    <small>
      Younger Futhark runes were used from 8th to 12th centuries in Scandinavia and their overseas settlements
    </small>
  </p>
  <div
    className="abbreviations"
  >
    <div
      className="column"
    >
      <h4>
        Abbreviations used:
      </h4>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            f.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          feminine.
        </dd>
      </dl>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            n.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          neuter.
        </dd>
      </dl>
    </div>
    <div
      className="column"
    >
      <h4>
        Works & Authors cited:
      </h4>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            Björn
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          Biörn Halldórsson.
        </dd>
      </dl>
    </div>
    <a
      className="link"
      href="/source-list"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      ➞ See all works cited in the dictionary
    </a>
  </div>
  <br />
  <h4>
    Also available in related dictionaries:
  </h4>
  <p>
    This headword also appears in dictionaries of other languages descending from Old Norse.
  </p>
  <ul
    className="list"
  >
    <li
      className="listItem"
    >
      <a
        href="https://old-icelandic.vercel.app/word/fadir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Icelandic - Geir Zoëga's Dictionary
      </a>
    </li>
    <li
      className="listItem"
    >
      <a
        href="https://old-norwegian-dictionary.vercel.app/word/fadir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Norwegian - Johan Fritzner's Dictionary
      </a>
    </li>
    <li
      className="listItem"
    >
      <a
        href="https://old-swedish-dictionary.vercel.app/word/fadhir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Swedish - K.F Söderwall's Dictionary
      </a>
    </li>
  </ul>
</article>
`;

exports[`WordDefinition component Matches snapshot 1`] = `
<article
  className="section"
>
  <header>
    <h1
      lang="non"
    >
      Af-burðr
    </h1>
    <small
      className="subHeading"
    >
      Old Norse Dictionary - 
      af-burðr
    </small>
    <p>
      Meaning of Old Norse word 
      <em>
        "
        af-burðr
        "
      </em>
       in English.
    </p>
  </header>
  <p>
    As defined by the Cleasby & Vigfusson Old Norse to English dictionary:
  </p>
  <p>
    <dfn
      className="capitalize"
    >
      af-burðr
    </dfn>
     Old Norse word can mean:
  </p>
  <dl
    className="definitionList"
  >
    <dt>
      <strong>
        af-burðr
      </strong>
       
    </dt>
    <dd
      className="itemDescription"
      dangerouslySetInnerHTML={
        {
          "__html": "m. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Ork<abbr title="neuter.">n.</abbr> 464.",
        }
      }
    />
  </dl>
  <dl
    className="definitionList"
  >
    <dt>
      <strong>
        af-burðr
      </strong>
       
    </dt>
    <dd
      className="itemDescription"
      dangerouslySetInnerHTML={
        {
          "__html": "<strong>II.</strong> ge<abbr title="neuter.">n.</abbr> sing. and pl. <strong>afburðar-, a-,</strong> freq. used as a prefix in some COMPDS with the notion of <i>gloriously, with distinction.</i> <strong>afburðar-digr,</strong> adj. <i>very thick</i>, biðr. 24. <strong>afburða-fræknligr,</strong> adj. <i>very gallant,</i> Ísl. ii. 369. <strong>afburðar-járn,</strong> <abbr title="neuter.">n.</abbr> <i>excellent iron,</i> Fms. x. 173. <strong>afburðar-maðr,</strong> m. <i>a man of mark,</i> Rb. 316, Ork<abbr title="neuter.">n.</abbr> 474, Grett. 133, Finnb. 318. <strong>afburðar-mikill,</strong> adj. <i>conspicuous,</i> Fms. v. 181. <strong>afburðar-skip,</strong> <abbr title="neuter.">n.</abbr> <i>a fine ship,</i> Fas. iii. 106. <strong>afburðar-vel,</strong> adv. <i>very well</i>, Hkr. ii. 265, Fms. ix. 515. <strong>afburðar-vænn,</strong> adj. <i>very fine,</i> Fas. i. 182.",
        }
      }
    />
  </dl>
  <p>
    Possible runic inscription in 
    <em>
      Younger Futhark
    </em>
    :
    <span
      className="rune"
    >
      ᛅᚠ-ᛒᚢᚱᚦᚱ
    </span>
    <br />
    <small>
      Younger Futhark runes were used from 8th to 12th centuries in Scandinavia and their overseas settlements
    </small>
  </p>
  <div
    className="abbreviations"
  >
    <div
      className="column"
    >
      <h4>
        Abbreviations used:
      </h4>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            f.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          feminine.
        </dd>
      </dl>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            n.
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          neuter.
        </dd>
      </dl>
    </div>
    <div
      className="column"
    >
      <h4>
        Works & Authors cited:
      </h4>
      <dl
        className="wrap"
      >
        <dt
          className="abbreviation"
        >
          <strong>
            Björn
          </strong>
        </dt>
        <dd
          className="explanation"
        >
          Biörn Halldórsson.
        </dd>
      </dl>
    </div>
    <a
      className="link"
      href="/source-list"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      ➞ See all works cited in the dictionary
    </a>
  </div>
  <br />
  <h4>
    Also available in related dictionaries:
  </h4>
  <p>
    This headword also appears in dictionaries of other languages descending from Old Norse.
  </p>
  <ul
    className="list"
  >
    <li
      className="listItem"
    >
      <a
        href="https://old-icelandic.vercel.app/word/fadir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Icelandic - Geir Zoëga's Dictionary
      </a>
    </li>
    <li
      className="listItem"
    >
      <a
        href="https://old-norwegian-dictionary.vercel.app/word/fadir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Norwegian - Johan Fritzner's Dictionary
      </a>
    </li>
    <li
      className="listItem"
    >
      <a
        href="https://old-swedish-dictionary.vercel.app/word/fadhir"
        rel="noopener noreferrer"
        target="_blank"
      >
        Old Swedish - K.F Söderwall's Dictionary
      </a>
    </li>
  </ul>
</article>
`;
